autokeras/auto/auto_model.py

```diff
@@ -28,17 +28,27 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.inputs = []
         self.outputs = []
+        self._hypermodels = []
+        self._nodes = []
+        self._built = False
         self.tuner = tuner.SequentialRandomSearch(self, objective=self._get_metrics())
 
     def build(self, hp):
         raise NotImplementedError
 
+    def _hyper_build(self):
+        self._built = True
+
     def fit(self,
             x=None,
             y=None,
             validation_data=None,
             trials=None,
             **kwargs):
+
+        if not self._built:
+            self._hyper_build()
+
```

Comment:
Should we set `self._built` here? Can the use override `self._hyper_build()` / `self.build()` and maybe not call `self.built = True`?

Reply:
I am not sure how to change it. We can discuss during our meeting tomorrow.